scrape/scrape.go

```diff
@@ -133,22 +133,96 @@ var (
 			Help: "How many times a scrape cache was flushed due to getting big while scrapes are failing.",
 		},
 	)
+	targetMetadataCache = newMetadataMetricsCollector()
 )
 
+// MetadataMetricsCollector is a Custom Collector for the metadata cache metrics.
+type MetadataMetricsCollector struct {
+	CacheEntries *prometheus.Desc
+	CacheBytes   *prometheus.Desc
+
+	scrapeManager *Manager
```

Comment:
Manager-related code goes in manager.go

Reply:
The way I see it manager is a dependency, the metrics are coming out of the `scrapePool` / `scrapeCache` so I decided to put it closer to those. 

Happy to move it, if you still think this belongs in the manager.

Reply:
It belongs in the manager imho

Reply:
moved it.